sympy/vector/tests/test_coordsysrect.py

```diff
@@ -390,3 +390,24 @@ def test_rotation_trans_equations():
     assert a._rotation_trans_equations() == (a.x, a.y, a.z)
     b = a.orient_new_axis('b', 0, -a.k)
     assert b._rotation_trans_equations() == (b.x, b.y, b.z)
+
+
+def test_transformation_composition():
+    a = CoordSys3D('a')
+    eq1 = a._rotation_trans_equations()
+    eq2 = a._transformation_equations()
+    assert a._transformation_composition(eq1, eq2) == (a.x, a.y, a.z)
+    from sympy import symbols
+    q0 = symbols('q0')
+    b = a.orient_new_axis('b', q0, a.k)
+    eq1 = b._transformation_equations()
+    eq2 = b._rotation_trans_equations()
+    assert b._transformation_composition(eq1, eq2) ==\
+           (sin(q0)*b.y + cos(q0)*b.x, -sin(q0)*b.x + cos(q0)*b.y, b.z)
+    b._connect_to_standard_cartesian('spherical')
```

Comment:
> Well, but by the way we define cylindrical and spherical, they are always aligned the same way.

The are aligned along the same direction, but we can still rotate them. Our cylindrical coordinate system can be rotated by rotation matrix.
I think that it's the easiest way to define whole family of coordinate system. From one standard system we can generate everyone else.

> Maybe we should define the transformations in the constructor of CoordSys3D and create a method analogous to orient_new for the generic transformations. What do you think?

I think that it's the best possible solution. I'd like to adapt the methods which we have already merged to do that. We can just rearrange them by moving some code to constructor.
I'm aware of creating too big constructor.